src/dbnode/storage/limits/query_limits_test.go

```diff
@@ -173,23 +183,47 @@ func TestLookbackLimit(t *testing.T) {
 
 			require.Equal(t, int64(0), limit.current())
 			verifyMetrics(t, scope, name, 0, 0, 17, exceededCount)
+
+			limit.reset()
+
+			opts.Limit = 0
+			require.NoError(t, limit.Update(opts))
+
+			exceededCount += verifyLimit(t, limit, 0, opts.Limit, test.forceExceeded)
+			require.Equal(t, int64(0), limit.current())
+
+			opts.Limit = 2
+			require.NoError(t, limit.Update(opts))
+
+			exceededCount += verifyLimit(t, limit, 1, opts.Limit, test.forceExceeded)
+			require.Equal(t, int64(1), limit.current())
+			verifyMetrics(t, scope, name, 1, 0, 18, exceededCount)
+
+			exceededCount += verifyLimit(t, limit, 1, opts.Limit, test.forceExceeded)
+			require.Equal(t, int64(2), limit.current())
+			verifyMetrics(t, scope, name, 2, 0, 19, exceededCount)
+
+			exceededCount += verifyLimit(t, limit, 1, opts.Limit, test.forceExceeded)
+			require.Equal(t, int64(3), limit.current())
+			verifyMetrics(t, scope, name, 3, 0, 20, exceededCount)
 		})
 	}
 }
 
-func verifyLimit(t *testing.T, limit *lookbackLimit, inc int, expectedLimit int64) int64 {
+func verifyLimit(t *testing.T, limit *lookbackLimit, inc int, expectedLimit int64, forceExceeded bool) int64 {
 	var exceededCount int64
 	err := limit.Inc(inc, nil)
-	if limit.current() <= expectedLimit || expectedLimit == 0 {
+	if (limit.options.Limit == 0 || limit.current() < expectedLimit) && !forceExceeded {
```

Comment:
This used to test if `expectedLimit == 0`, rather than `limit.options.Limit == 0`; same with the other one. Should be equivalent if the code is correct, but differs from previous behaviour here

Reply:
Switched back